test/Microsoft.ML.Tests/Transformers/TextFeaturizerTests.cs

```diff
@@ -51,6 +52,95 @@ public void TextFeaturizerWithPredefinedStopWordRemoverTest()
             Assert.Equal("stop words", string.Join(" ", prediction.OutputTokens));
         }
 
+        [Fact]
+        public void TextFeaturizerWithWordFeatureExtractorTest()
+        {
+            var data = new[] { new TestClass() { A = "This is some text in english", OutputTokens=null},
+                               new TestClass() { A = "This is another example", OutputTokens=null } };
+            var dataView = ML.Data.LoadFromEnumerable(data);
+
+            var options = new TextFeaturizingEstimator.Options()
+            {
+                WordFeatureExtractor = new WordBagEstimator.Options() { NgramLength = 1 },
+                CharFeatureExtractor = null,
+                Norm = TextFeaturizingEstimator.NormFunction.None,
+                OutputTokensColumnName = "OutputTokens"
+            };
+            var pipeline = ML.Transforms.Text.FeaturizeText("Features", options, "A");
+            var model = pipeline.Fit(dataView);
+            var engine = model.CreatePredictionEngine<TestClass, TestClass>(ML);
+
+            var prediction = engine.Predict(data[0]);
+            Assert.Equal("this is some text in english", string.Join(" ", prediction.OutputTokens));
+            var expected = new float[] { 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 0.0f, 0.0f };
+            Assert.Equal(expected, prediction.Features);
+
+            prediction = engine.Predict(data[1]);
+            Assert.Equal("this is another example", string.Join(" ", prediction.OutputTokens));
+            expected = new float[] { 1.0f, 1.0f, 0.0f, 0.0f, 0.0f, 0.0f, 1.0f, 1.0f };
+            Assert.Equal(expected, prediction.Features);
+        }
+
+        [Fact]
+        public void TextFeaturizerWithCharFeatureExtractorTest()
+        {
+            var data = new[] { new TestClass() { A = "abc efg", OutputTokens=null},
+                               new TestClass() { A = "xyz", OutputTokens=null } };
+            var dataView = ML.Data.LoadFromEnumerable(data);
+
+            var options = new TextFeaturizingEstimator.Options()
+            {
+                WordFeatureExtractor = null,
+                CharFeatureExtractor = new WordBagEstimator.Options() { NgramLength = 1 },
+                Norm = TextFeaturizingEstimator.NormFunction.None,
+                OutputTokensColumnName = "OutputTokens"
+            };
+            var pipeline = ML.Transforms.Text.FeaturizeText("Features", options, "A");
+            var model = pipeline.Fit(dataView);
+            var engine = model.CreatePredictionEngine<TestClass, TestClass>(ML);
+
+            var prediction = engine.Predict(data[0]);
+            Assert.Equal("abc efg", string.Join(" ", prediction.OutputTokens));
+            var expected = new float[] { 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 0.0f, 0.0f, 0.0f };
```

Comment:
>1.0f [](start = 89, length = 4)

should this be 0? or is this the end marker?

Reply:
Its the end marker.

---
In reply to: [267967436](https://github.com/dotnet/machinelearning/pull/3036#discussion_r267967436) [](ancestors = 267967436)